src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -264,31 +288,30 @@ def _alloc_nompi(self, cud):
         Find a suitable set of cores and gpus *within a single node*.
 
         Input:
-        cud: Compute Unit description. Needs to specify at least one process,thread
-        or GPU process.
+        cud: Compute Unit description. Needs to specify at least one CPU 
+        process and one thread per CPU process, or GPU process.
 
         '''
 
-        # Get the number of processes, threads of GPU processes this unit requests.
-        # For threads, if the entry is not specified in the description, then 
-        # assume it is one thread per process.
+        # Get the number of CPU processes, threads per CPU process 
+        # and, GPU processes this unit requests.
         requested_procs  = cud['cpu_processes']
-        threads_per_proc = cud.get('cpu_threads',1)
+        threads_per_proc = cud['cpu_threads']
         requested_gpus   = cud['gpu_processes']
```

Comment:
Haha, this is funny - you change back an edit from Iannis, but do not add back a check which was removed by Iannis, too, which relates to that change.  The consequence of that I'll leave as an exercise to the reader  :-D

Reply:
I did notice the check Giannis removed. But isn't the default number of cpu_threads=1? https://github.com/radical-cybertools/radical.pilot/blob/feature/gpu/src/radical/pilot/compute_unit_description.py#L364

Reply:
The check that Giannis removed checks if cpu_threads is None (or 0), both of which the user has to manually set (since the default is 1). At which point he/she deserves an error! :P No?

Reply:
Indeed, the default is such.  But I found myself sometimes specifying `0` in the sense of `no additional threads` - which is not what the user should do, but there is also no harm in catching that obvious bad value and replacing it with a sensible `1`, right?

Reply:
Sorry can you elaborate on no ```additional``` threads. Once you set cpu_threads=0, the number of cores acquired for that process is 0 (=no. of process * threads per process). Do you mean, you set it to 0 by mistake?

Reply:
Yes, by mistake.  I guess that points to a different issue in the API really, which may benefit from a different discussion.  Look, in the API, we specify `cpu_processes` and `cpu_threads`, but we interpret the latter as `threads_per_process`, which is not quite the same, right?  So, in the API, `0` actually makes semantic sense.

Happy to open up a discussion about the API naming - but I'd rather do that independent from the scheduler discussion, if that's ok.  But do you see how a `0` could well happen to be specified by a user?

Reply:
I think I understand what you meant. Please see my comment in the previous change request of this PR.